src/main/thrift/parquet.thrift

```diff
@@ -223,6 +223,17 @@ struct Statistics {
     */
    5: optional binary max_value;
    6: optional binary min_value;
+   /** The number of bytes the row/group or page would take if encoded with plain-encoding */
+   7: optional i64 plain_encoded_bytes;
+   /** 
+     * When present there is expected to be one element corresponding to each repetition (i.e. size=max repetition_leve) 
+     * where each element represens the count of the number of times that level occurs in the page/column chunk.
+     */
+   8: optional list<i64> repetition_level_histogram;
```

Comment:
Seems it's a per-rowgroup statistics?

Reply:
I'm not sure I understand, I thought statistics can be on either page or row group.

Reply:
Oh, sorry for missunderstand this. How can we make full use of this histogram?

Reply:
These levels are rather complicated for high-level applications, we may need to provide some utility to translate to norms (e.g. struct/map/array/null) that are much easier to understand.

Reply:
I agree there is complexity here but I think this is the simplest and most complete set of information that we can provide readers.  I think the utility methods would likely be per system but we could likely provide some in core packages that give leaf value estimates that follow the rules above.

@mapleFU in terms of usage these would be used similar to how row level reconstruction is done with them.  For instance total number of nulls at the leaf can be computed assuming no repeated fields as the cumulative sum of the first n-1 entries in definition level.  Similarly number of nested lists assuming no null lists should be computable with a similar cumulative sum.  When lists are nullable repetition and definition level need to be looked at together to determine null vs empty lists.  Similarly the number of empty lists at  level can be inferred by looking at definition and repetition levels together

Reply:
I guess we could provide utility methods that account for common permutations of memory models

Reply:
Seems It can help pushdown some filter on List/Map, and helping constructing the list. It's great, but I think maybe we need some samples? Because it's a bit hard to understand how to make full use of it. Like some rules in <Storing and Querying Tree-Structured Records in Dremel> ?

Reply:
So some simple examples in from Arrow (my rep/def level calculations are rusty so please double check to make sure they make sense):
1.  If all you have are nested structures (say 3 nullable levels), then the number of null values at the leaf is the sum of the first 3 elements of the definition level histogram.  Individual level nullability for Arrow isn't super important because it always leaves the space for nulls so it is pretty much square.  You could however determine that there are no nulls at the second level of nesting by checking the appropriate histogram box (and potentially save the allocation of the bit vector in that case)
2.  If you have two nested lists and assuming lists and elements are not nullable.  The number of inner nested lists is `rep_hist[0]  - def_hist[0] + {rep_hist[1]` (outer list starts - empty outer lists + inner list starts).

Reply:
Sounds feasible. Then we are good to store level histograms separately.